File: CommonShareableTypes15.7/CodeFile1.cs
/*
 * creation date  06 oct 2020
 * last change    21 jun 2021
 * author         artur
 */
using System;

namespace CommonShareableTypes
{
    public interface IAppFunctionality  // IAppFunctionality - этот интерфейс будет служить полиморфной основой для всех оснасток, что могут
    {                                   //   быть применены расширяемым приложением
                                        // Другими словами, расширяемое приложение сможет пользоваться только теми оснатсками, что совместимы с
                                        //   этим интерфейсом
        void DoIt();
    }
    [AttributeUsage(AttributeTargets.Class)]
    public class CompanyInfoAttribute : System.Attribute  // CompanyInfo.. - этот атрибут будет описывать типы оснастки, храня инфо о месте
    {                                                     //   их происхождения (компании). С этим атрибутом также будут знакомы и расширяемое
                                                          //   приложение, и оснастки
        public string CompanyName { get; set; }
        public string CompanyUrl { get; set; }
    }
}

File: VbShareIn15.7/CodeFile1.vb
'
'  creation date  11 jun 2021
'  last change    21 jun 2021
'  author         artur
'
Imports CommonShareableTypes
' CommonShareableTypes - этот проект я привязал через Reference Manager (т.е. так, как это было с проектом CSharpSnareIn)
' Комментарии автоформатируются by VS, и я не нашёл решение этому

<CompanyInfo(CompanyName:="Chucky's Software", CompanyUrl:="www.ChuckySoft.com")>
Public Class VbSnareIn                                           ' <> - этим в VB помечаются атрибуты (как [] в C#)
    Implements IAppFunctionality                                 ' Implements - этим в VB помечаются интерфейсы перед типом
    Public Sub DoIt() Implements IAppFunctionality.DoIt
        Console.WriteLine("You have just used the VB snap-in!")  ' DoIt() - интересно, но как только я нажал на Tab для завершения записи
                                                                 '   IAppFunctionality, VS сама добавила загатовку для этого метода
    End Sub
End Class
